core/src/main/java/org/infinispan/commands/write/RemoveExpiredCommand.java

```diff
@@ -193,8 +196,8 @@ public int hashCode() {
 
    @Override
    public long getFlagsBitSet() {
-      // Override the flags
-      return FlagBitSets.SKIP_CACHE_LOAD;
+      // Override the flags to always include skip cache load
+      return flags | FlagBitSets.SKIP_CACHE_LOAD;
```

Comment:
Maybe it would be nicer to add the `SKIP_CACHE_LOAD` flag in `CacheImpl`

Reply:
Hrmm, okay. Let me try to reorg it.